File: backend/Libary/Model/Project/Project.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Libary.Model.Project
{
    public class Project
    {
        public int Id { get; set; }
        public string OwnerId { get; set; } // A bejelentkezett felhasználó Identity ID-ja
        public string Title { get; set; }
        public string? Description { get; set; }
        public DateTime? Deadline { get; set; }
        public string Status { get; set; } = "ongoing"; // ongoing, archived, invited

        // Kapcsolatok
        public ICollection<ProjectMember> Members { get; set; } = new List<ProjectMember>();
        public ICollection<ProjectTask> Tasks { get; set; } = new List<ProjectTask>();
    }
}

File: frontend/src/types/project.ts
// src/types/project.ts

export interface ProjectMember {
  userId: string;
  name: string;
  role: string;
  avatar?: string;
}

export interface Todo {
  id: number;
  text: string;
  amount: number | null;
  completed: boolean;
  workerId: string | null;
}

export interface Task {
  id: number;
  projectId: number;
  title: string;
  description?: string;
  amount: number | null;
  deadline?: string;
  status: 'todo' | 'in-progress' | 'done';
  assignedTo: string[];
  todos: Todo[];
}

export interface Project {
  id: number;
  ownerId: string;
  title: string;
  description?: string;
  deadline?: string;
  status: 'ongoing' | 'archived' | 'invited';
  members: ProjectMember[];
  tasks: Task[];
}

export interface CalendarEntry {
  id: number;
  title: string;
  description?: string;
  date: string;
  isEntry: boolean; // True, ha ez egy saját naptár bejegyzés
}
